octodns/provider/transip.py

```diff
@@ -39,315 +38,229 @@ class TransipProvider(BaseProvider):
         }
 
-    def _data_for_SSHFP(self, _type, records):
-        _values = []
-        for record in records:
-            algorithm, fp_type, fingerprint = record['content'].split(' ', 2)
-            _values.append({
-                'algorithm': algorithm,
-                'fingerprint': fingerprint.lower(),
-                'fingerprint_type': fp_type
-            })
-
+    def format_srv(record):
+        priority, weight, port, target = record.content.split(" ", 3)
         return {
-            'type': _type,
-            'ttl': self._get_lowest_ttl(records),
-            'values': _values
+            "port": port,
+            "priority": priority,
+            "target": _parse_to_fqdn(target, current_zone),
+            "weight": weight,
         }
 
-    def _data_for_CAA(self, _type, records):
-        _values = []
-        for record in records:
-            flags, tag, value = record['content'].split(' ', 2)
-            _values.append({
-                'flags': flags,
-                'tag': tag,
-                'value': value
-            })
-
+    def format_sshfp(record):
+        algorithm, fp_type, fingerprint = record.content.split(" ", 2)
         return {
-            'type': _type,
-            'ttl': self._get_lowest_ttl(records),
-            'values': _values
+            "algorithm": algorithm,
+            "fingerprint": fingerprint.lower(),
+            "fingerprint_type": fp_type,
         }
 
-    def _data_for_TXT(self, _type, records):
-        _values = []
-        for record in records:
-            _values.append(record['content'].replace(';', '\\;'))
-
-        return {
-            'type': _type,
-            'ttl': self._get_lowest_ttl(records),
-            'values': _values
-        }
+    def format_caa(record):
+        flags, tag, value = record.content.split(" ", 2)
+        return {"flags": flags, "tag": tag, "value": value}
+
+    def format_txt(record):
+        return record.content.replace(";", "\\;")
+
+    value_formatter = {
+        "MX": format_mx,
+        "SRV": format_srv,
+        "SSHFP": format_sshfp,
+        "CAA": format_caa,
+        "TXT": format_txt,
+    }.get(type_, lambda r: r.content)
+
+    return {
+        "type": type_,
+        "ttl": _get_lowest_ttl(records),
+        "values": [value_formatter(r) for r in records],
+    }
+
+
+def _parse_to_fqdn(value, current_zone):
+    # TransIP allows '@' as value to alias the root record.
+    # this provider won't set an '@' value, but can be an existing record
+    if value == TransipProvider.ROOT_RECORD:
+        value = current_zone.name
+
+    if value[-1] != ".":
+        value = "{}.{}".format(value, current_zone.name)
+
+    return value
+
+
+def _get_lowest_ttl(records):
+    return min([r.expire for r in records] + [100000])
+
+
+def _entries_for(name, record):
+    values = record.values if hasattr(record, "values") else [record.value]
+    formatter = {
+        "MX": lambda v: f"{v.preference} {v.exchange}",
```

Comment:
Per https://github.com/octodns/octodns/issues/711 f-strings are allowed as of formal python 2 deprecation aug 16th so these are now ok.